clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -169,10 +171,18 @@
             " the message was sent to an incompatible broker. See the broker logs for more details.")),
     UNSUPPORTED_FOR_MESSAGE_FORMAT(43,
         new UnsupportedForMessageFormatException("The message format version on the broker does not support the request.")),
-    POLICY_VIOLATION(44, new PolicyViolationException("Request parameters do not satisfy the configured policy.")),
-    OUT_OF_ORDER_SEQUENCE_NUMBER(45, new OutOfOrderSequenceException("The broker received an out of order sequence number")),
-    DUPLICATE_SEQUENCE_NUMBER(46, new DuplicateSequenceNumberException("The broker received a duplicate sequence number")),
-    PRODUCER_FENCED(47, new ProducerFencedException("Producer attempted an operation with an old epoch"));
+    POLICY_VIOLATION(44,
+        new PolicyViolationException("Request parameters do not satisfy the configured policy.")),
+    OUT_OF_ORDER_SEQUENCE_NUMBER(45,
+        new OutOfOrderSequenceException("The broker received an out of order sequence number")),
+    DUPLICATE_SEQUENCE_NUMBER(46,
+        new DuplicateSequenceNumberException("The broker received a duplicate sequence number")),
+    INVALID_PRODUCER_EPOCH(47,
+        new ProducerFencedException("Producer attempted an operation with an old epoch")),
```

Comment:
Not sure if this is clear, but all classes under `errors` are public API. As such, we should consider the naming of such classes carefully. It would be good to take a pass at all the names before the release and make sure they still make sense.